counsel.el

```diff
@@ -1237,15 +1234,15 @@ Typical value: '(recenter)."
   :type 'hook
   :group 'ivy)
 
-(defun counsel-git-grep-function (str &optional _pred &rest _unused)
-  "Grep in the current git repository for STRING."
+(defun counsel-git-grep-function (str repo-size)
```

Comment:
I wonder whether we should worry about backward compatibility here, given the function plays a central role and was never given an internal name such as `counsel--git-grep-fn`?

Reply:
what do you have in mind when talking about backward compatibility?

Reply:
@mookid I'm thinking of external libraries and personal customisations which may be using `counsel-git-grep-function` in their own code. The previous variadic arglist comprised one mandatory argument, whereas the new arglist is no longer variadic and comprises two mandatory arguments.

We can still support the old calling convention by making `REPO-SIZE` optional and use an [`advertised-calling-convention` declaration](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declare-Form.html) to advertise the new API.